model/multi_gases.F90

```diff
@@ -17,6 +17,9 @@
 !* You should have received a copy of the GNU Lesser General Public
 !* License along with the FV3 dynamical core.
 !* If not, see <http://www.gnu.org/licenses/>.
+!* --------- Default gas values for wam ----------------------------------------
+!* R_n2+= 295.3892, R_h2o= 461.50, R_o3=173.2247, R_o= 519.674, R_o2=259.8370
+!* Cpn2+=1031.1083, Cph2o=1846.00, Cpo3=820.2391, Cpo=1299.185, Cpo2=918.0969
```

Comment:
These lines need to be moved outside of the license block.

Reply:
It is okay to use if-clauses though MULTI_GASES and MOLECULAR_DIFFUSION are designed specific for WAM.  For general use, some codes in molecular diffusion to compute diffusion coefficients need to add more general gases through multi_gases namelist.  Since MD and MG have lots ifdef, I suggest we leave them for next PR to change to if-clauses.

Reply:
Thanks to point out locations to  change, we can fix them.

Reply:
> I provide this review with the caveat that @lharris4 will need to weigh in as well.
> 
> There is no need to have all the #ifdefs in atmosphere.F90. The variable "time_total" is already an optional argument to fv_dynamics and the way it is used as it traverses the call tree will not impact anything if it is "always present" in the UFS top level atmospheric dynamics driver.

#ifdef for time_total is removed in atmosphere.F90
> 
> If molecular diffusion is added as a runtime option, as suggested by @lharris4, many of the #ifdefs can be removed and handled with standard if-clauses.

Please suggest if it is OK to keep it is a compiling option.
> 
> The molecular diffusion logic in dyn_core.F90 (1256-1405) and fv_dynamics.F90 (771-829) and supporting logic should be encapsulated in subroutines which should live in the newly added molecular_diffusion.F90. In the current setup with #ifdefs, the routines can be simple stubs in the absence of the macro.

I guess this depends on if #ifdef is changed, right?

> 
> id_ow and id_o2w are only used in fv_diagnostics.F90 and should be defined in fv_diagnostics.h and not the diag_type within fv_arrays.F90.
> 
id_ow and id_o2w are defined in fv_diagnostics.h  as id_o3w.

> Please fix nh_utils.F90, sw_core.F90, tp_core.F90 and external_ic.F90 so we can see only the true diffs and not the whole file as a diff.
The format difference is fixed for the above files.

Reply:
> These lines need to be moved outside of the license block.

revised